src/Microsoft.Framework.PackageManager/Restore/NuGet/NuGetv2Feed.cs

```diff
@@ -232,17 +240,49 @@ public PackageInfo BuildModel(string id, XElement element)
             });
         }
 
+        private void EnsureValidFindPackagesResponse(XDocument doc)
+        {
+            if (!string.Equals(doc.Root.GetDefaultNamespace().ToString(), _defaultNamespace,
+                StringComparison.OrdinalIgnoreCase))
+            {
+                throw new FormatException("The XML is not a valid NuGet v2 service response to FindPackagesById() request");
```

Comment:
I see. Then I'll use `InvalidOperationException` instead. Thanks for the detailed explanation!